ast/parser.go

```diff
@@ -644,9 +644,12 @@ func (p *Parser) parseRules() []*Rule {
 	}
 
 	if p.s.tok == tokens.Else {
-
+		if r := rule.Head.Ref(); len(r) > 1 && !r[len(r)-1].Value.IsGround() {
```

Comment:
For a non-ref-head rule, like
```rego
a := 1 if true
```

the ref is `Ref{VarTerm("a")}`, and that would be a false-positive for the check if done without `len(r)>1`.